packages/amplify-cli-core/src/state-manager/stateManager.ts

```diff
@@ -72,9 +75,129 @@ export class StateManager {
       ...options,
     };
 
-    return this.getData<$TSTeamProviderInfo>(filePath, mergedOptions);
+    return this.getData<CustomIAMPolicies>(filePath, mergedOptions);
   };
 
+  getCustomPolicies = (categoryName: string, resourceName: string): any => {
+    const filePath = pathManager.getCustomPoliciesPath(categoryName, resourceName);
+    let customPolicies: CustomIAMPolicies = {
+      policies :  []
+    }
+    const data = JSONUtilities.readJson<CustomIAMPolicies>(filePath, {throwIfNotExist : false});
+    const ajv = new Ajv();
+    const { envName } = this.getLocalEnvInfo()
+
+    //validate if the policies match the custom IAM policies schema, if not, then not write into the CFN template
+    const validatePolicies = ajv.compile(CustomIAMPoliciesSchema);
+    if (!data || !validatePolicies(data)) {
+      printer.warn(`The format of custom IAM policies in the ${categoryName} ${resourceName} is not valid`);
+      return undefined;
+    }
+    const validatePolicy = ajv.compile(CustomIAMPolicySchema);
+    for (const policy of data.policies) {
+      if (validatePolicy(policy)) {
+        if (!policy.Effect) policy.Effect = 'Allow';
+        const policyWithoutEnv = this.replaceEnvForCustomPolicies(policy, envName);
+        customPolicies.policies.push(policyWithoutEnv);
+      }
+      else {
+        printer.warn(`The format of custom IAM policies in the ${categoryName} ${resourceName} is not valid`);
+        return undefined;
+      }
+    }
+    //the policies with env will be written to the custom policies file again
+    JSONUtilities.writeJson(filePath, data);
```

Comment:
Why are you writing the policies back?

Reply:
This will show up as modified when switch an environment

Reply:
The customers can see the resource with current env

Reply:
I am not sure I understand what you mean by thsi?